docker-compose.yml

```diff
@@ -433,6 +433,24 @@ services:
     volumes:
       - oracle:/opt/oracle/data
 
+  exasol:
+    build: ./docker/exasol
+    image: ibis-exasol
+    privileged: true
```

Comment:
Is this required? If so, can you add a short comment about why?

Reply:
Looked into this a bit.

In theory we should be able to use `cap_add` to specify the list of [capabilities](https://man7.org/linux/man-pages/man7/capabilities.7.html) the container needs, but in practice the container appears to be using some unnecessary patterns that require `privileged: true` (which gives more privileges than can be specified in `cap_add`)

It looks like the container is trying to write to hugepages using the proc filesystem to verify that hugepages can be allocated. I don't think this is necessary, you should be able to allocate hugepages without needing to use the proc filesystem to check whether it's possible to do so.

The `IPC_LOCK` capability enables this:

![image](https://github.com/ibis-project/ibis/assets/417981/1bee2fdb-33c2-4c3f-bb25-c7299b46927f)

This isn't a blocker, be we should work with whoever created this image to not require `privileged: true` to run it.

Reply:
**TL;DR:** I agree that one does not want the `privileged` flag to be set. Sadly, afik currently it is not possible to run the container without it.

## Details
To the best of my knowledge, there have already been discussions around this issue. However, I'm not sure if there has been any "final" outcome. I'll take your feedback as an opportunity to reignite the conversation about it. I will also take a closer look at how I can contribute to resolving this issue.

Reply:
Happy to help in whatever way is needed. Not a blocker for this PR in any case. We can try to resolve this in parallel.

Reply:
@cpcloud I took some time and was talking to various people regarding the matter of the `--privileged` flag.

The TL;DR is:

* It is known
* It is rated as not ideal (should change)
* The huge pages issue is not the major reason why removing it
  would require a significant amount of effort
* It may be done in future, but there is no time table on it

Reply:
Hi @cpcloud,

currently I try to upstream a health-check to the official `exasol docker image`. As of now this mostly looks like the code snippet(s) bellow. That said, from what I understood it is very likely that it only will be part of  `exasol docker images` for  versions `>= 8`.

So I was wondering if it would make sense, to reintroduce a "custom" docker image inhering from the exasol one. But this time I would just add the `healthcheck` bellow.  Building such a container should not have much overhead, compared to fetching the "official" one. Also by adapting this, we would have a somewhat "streamlined" `healthcheck` for the exasol DB. 

Just want to get your take on it, please let me now what your preferred solution is (current vs image with health script).

```python
#! /usr/bin/env python3
"""
This script implments a simple health check inteded for use as dockers health check.

For futher details regarding the docker health check,
see also https://docs.docker.com/engine/reference/builder/#healthcheck.
"""
from __future__ import annotations
import sys
import shutil
import argparse
from enum import IntEnum
from pathlib import Path
from subprocess import run
from functools import partial, wraps


_stdout = print
_stderr = partial(print, file=sys.stderr)


class Status(IntEnum):
    Healthy = 0
    Unhealthy = 1
    Reserved = 2


class BinaryNotFound(Exception):
    pass


def _find_binary(name: str, basepath: str | Path) -> Path:
    """
    Searches for a binary within the current $PATH and recursivly in the specified directory

    Search Order:
        1. current $PATH
        2. specified fs path

    Args:
        name: of the binary to search for.
        basepath: to search if the binary isn't found in the current $PATH.


    Return:
        Path to the first binary which matches the search criteria.


    Raises:
        BinaryNotFound: The specified binary couldn't be found.
        
    """
    basepath = Path(basepath)

    binary = shutil.which(name)
    if binary is not None and binary != '':
        return Path(binary)

    try:
        binary = next(Path(basepath).rglob(name))
    except StopIteration:
        raise BinaryNotFound(f"Could not find  binary [{name}].")

    return Path(binary)
    

def _health_check(username, password, host, port) -> Status:
    try:
        exaplus = _find_binary("exaplus", "/")
    except BinaryNotFound as ex:
        _stderr(f"{ex}")
        return Status.Unhealthy

    result = run([
        exaplus,
        "-c",
        f"{host}:{port}",
        "-u",
        f"{username}",
        "-p",
        f"{password}",
        "-encryption",
        "OFF",
        "-sql",
        "'SELECT 1;'"
    ], capture_output=True)
    status = Status.Healthy if result.returncode == 0 else Status.Unhealthy 

    return status


def main(argv=None):

    def _parser() -> ArgumentParser:
        parser = argparse.ArgumentParser(
                formatter_class=argparse.ArgumentDefaultsHelpFormatter
        )
        parser.add_argument(
            '--host', type=str, default='127.0.0.1',
            help='to connect to'
        )
        parser.add_argument(
            '--port', type=int, default=8563,
            help='to connect to'
        )
        parser.add_argument(
            '--username', type=str, default='sys',
            help='to authenticate against the db'
        )
        parser.add_argument(
            '--password', type=str, default='exasol',
            help='to authenticate against the db'
        )
        parser.add_argument(
            '--debug', action="store_true", default=False,
            help="disable the 'last resort' exception handler"
        )
        return parser

    def _main(username, password, host, port) -> Status:
        exaplus = _find_binary("exaplus", "/usr/opt")
        return _health_check(username, password, host, port)

    def _protect(f, *args, **kwargs) -> Status:
        
        @wraps(f)
        def wrapper(*args, **kwargs):
            try:
                return f(*args, **kwargs)
            except Exception as ex:
                _stdout(f"Error: {ex}")
                return Status.Unhealthy

        return wrapper

    parser = _parser()
    args = parser.parse_args(argv)

    entry_point = _main
    if not args.debug:
        entry_point = _protect(entry_point)

    return entry_point(
        args.username,
        args.password,
        args.host,
        args.port
    )


if __name__ == '__main__':
    sys.exit(main())
```

```Dockerfile
...
ADD healthcheck.py healthcheck.py
HEALTHCHECK --interval=10s --retries=10 --timeout=90s CMD python3 healthcheck.py
```

Reply:
I'll leave the `health-check` in this PR as is. Up-streaming it into the official container is still ongoing.
Still it is already clear that it only will be part of version(s) `>= 8.x.y`. Therefore, I'll address the change of the `health-check` when upgrading the DB-Backend version.